alacritty_terminal/src/grid/storage.rs

```diff
@@ -832,8 +837,15 @@ mod test {
             visible_lines: Line(0),
             len: 9,
         };
-        assert_eq!(storage.inner, shrinking_expected.inner);
-        assert_eq!(storage.zero, shrinking_expected.zero);
+        assert_eq!(
+            storage.inner[..shrinking_expected.zero],
+            shrinking_expected.inner[..shrinking_expected.zero]
+        );
+        assert_eq!(
+            storage.inner[storage.zero..],
+            shrinking_expected.inner[shrinking_expected.zero..]
+        );
+        assert_eq!(storage.zero, MIN_INIT_SIZE + 2);
         assert_eq!(storage.len, shrinking_expected.len);
```

Comment:
I'm not sure about zero thing in this test now tbh.

Reply:
First of all I think there has been a copy-paste error here? It shouldn't be called `shrinking_expected`, but something different like `initialize_expected`, or probably better just `expected` or `expected_storage`.

I also think that we should probably modify the expected `Storage`, rather than our assertions. Having additional extra lines is part of our logic and thus part of what we should verify in our tests.

Lastly `MIN_INIT_SIZE + 2` as zero is a bit of a magic value, it should probably be `old_zero + MIN_INIT_SIZE` or something like that. Where `old_zero` is the previously defined initial `zero` offset before the `initialize` call.